internal/logger/default_logger.go

```diff
@@ -21,124 +21,122 @@ import (
+	pc, _, _, _ := runtime.Caller(logCallDepth)
+	return runtime.FuncForPC(pc).Name()
 }
 
-func (l *DefaultLogger) Warnf(format string, values ...interface{}) {
-	if l.canLogWarn() {
-		s := fmt.Sprintf("WARN : %s", fmt.Sprintf(format, values...))
-		l.Output(logCallDepth, s)
-	}
+// LogAdaptor is used to convert logger implementations of public interface logger.LogAdaptor to internal logging interface LogAdaptor
+type LogAdaptor struct {
+	logger.Logger
 }
 
-// Error logs the given arguments at error level if the level is greater than or equal to error level.
-func (l *DefaultLogger) Error(err error) {
-	l.Errorf(err.Error())
+// Debug runs the given function to generate the logger string, if logger level is debug or finer.
+func (la LogAdaptor) Debug(f func() string) {
+	la.Log(logger.DebugLevel, f)
 }
 
-func (l *DefaultLogger) Errorf(format string, values ...interface{}) {
-	if l.canLogError() {
-		s := fmt.Sprintf("ERROR: %s", fmt.Errorf(format, values...).Error())
-		l.Output(logCallDepth, s)
-	}
+// Trace runs the given function to generate the logger string, if logger level is trace or finer.
+func (la LogAdaptor) Trace(f func() string) {
+	la.Log(logger.TraceLevel, f)
 }
 
-func (l *DefaultLogger) findCallerFuncName() string {
-	pc, _, _, _ := runtime.Caller(logCallDepth)
-	return runtime.FuncForPC(pc).Name()
+// Info runs the given function to generate the logger string, if logger level is info or finer.
+func (la LogAdaptor) Info(f func() string) {
+	la.Log(logger.InfoLevel, f)
 }
 
-func (l *DefaultLogger) canLogTrace() bool {
-	return l.Level >= traceLevel
+// Infof formats the given string with the given values, if logger level is info or finer.
+func (la LogAdaptor) Infof(format string, values ...interface{}) {
+	la.Log(logger.InfoLevel, func() string {
+		return fmt.Sprintf(format, values...)
+	})
 }
 
-func (l *DefaultLogger) canLogInfo() bool {
-	return l.Level >= infoLevel
+// Warn formats the given string with the given values, if logger level is warn or finer.
+func (la LogAdaptor) Warn(f func() string) {
+	la.Log(logger.WarnLevel, f)
 }
 
-func (l *DefaultLogger) canLogWarn() bool {
-	return l.Level >= warnLevel
+// Warnf formats the given string with the given values, if logger level is warn or finer.
+func (la LogAdaptor) Warnf(format string, values ...interface{}) {
+	la.Log(logger.WarnLevel, func() string {
+		return fmt.Sprintf(format, values...)
+	})
 }
 
-func (l *DefaultLogger) canLogError() bool {
-	return l.Level >= errorLevel
+// Error logs the given args at error level.
+func (la LogAdaptor) Error(err error) {
+	la.Log(logger.ErrorLevel, func() string {
+		return err.Error()
+	})
 }
 
-func (l *DefaultLogger) CanLogDebug() bool {
-	return l.Level >= debugLevel
+// Errorf formats the given string with the given values, if logger level is error or finer.
+func (la LogAdaptor) Errorf(format string, values ...interface{}) {
+	la.Log(logger.ErrorLevel, func() string {
+		return fmt.Sprintf(format, values...)
```

Comment:
This `Error` method is different from the old one, which created an error with the given format and values.